consensus/ibft/snapshot.go

```diff
@@ -46,6 +46,35 @@ func (i *Ibft) setupSnapshot() error {
 		}
 	}
 
+	// If the snapshot is not found, or the latest snapshot belongs to a previous epoch,
+	// we need to start rebuilding the snapshot from the beginning of the current epoch
+	// in order to have all the votes and validators correctly set in the snapshot,
+	// since they reset every epoch.
```

Comment:
I understand that we dump old epochs anyway, just wanted to make sure if it's ok to have a gap of 1 epoch in the store, for example.

Reply:
It'll work as long as there is a snapshot to be used to calculate next snapshot. But restored snapshot must be at the beginning of the epoch. Subsequent codes advances snapshot from the beginning of the epoch to the current number.

The stored snapshot doesn't have to be in same epoch to the current. So that, second condition `metaEpoch < currentEpoch` is just for jumping to current epoch.